### ISSUE TITLE
kustomize based extensions

### ISSUE THREAD

**Issue by mikhail-sakhnov** [k0sproject/k0s#1525]
### Is your feature request related to a problem? Please describe.

There are helm based extensions now in the k0s. The way they work is basically installing helm releases.

It would be nice to have more or less the same mechanism for kustomize based extensions.

### Describe the solution you would like

Introduce new field in the config `spec.extensions.kustomize` which is the array of path to manifests with overlays. 
Each path is saved under the /var/lib/k0s/mainfests/kustomize/ (or any other stack applier location, up to consideration) after processing with kustomize.

### Additional context

Needs proper design first.

**Comment by chattytak** [comment]
To be honest, the current helm Extension is so buggy that I doubt it can withstand real production use. I feel that it is a beta level feature.
From the most recent issues I've seen, it looks like even the most basic bugs have gone unnoticed until now. Probably, this feature is not used by many people.
Or if you're happy using it in its current state, you're a light user.

Considering the current situation of helm extension, I don't think it is appropriate to reach out to Kustomize further from here.
It will divide development resources and the two features will become more and more delicate.
And as with the helm extension, I expect that people will use the standard kusutomize command and not use this proprietary extension.

There is no doubt that helm extention is in a difficult situation for everyone.
The first reason is that there is almost no documentation for this feature, even though it is unique to k0s.
If it is a helm standard feature, we can look for knowledge on the internet, but if it is a k0s proprietary feature, the answer is only here on GitHub.
Compared to other k8s distributions (minikube, microk8s, k3s . . etc.), k0s is still minor and information is scarce.
In addition, there are only a handful of people who are willing to try out features without official documentation and publish them on the Internet.
I think there are more people who are satisfied with the standard helm commands than trying to use the proprietary features that have little documentation.

This is a minor complaint about Helm Extention, but it increments the revision for all releases, even if some releases have not been updated and some have been updated. I can't decide if this is also the intended behavior or a bug.
In addition, I don't know where the logs are when I run helm, so if it doesn't become Deployed, it's the beginning of hell.
This is also due to the lack of documentation. I don't know if I'm the only one who doesn't know how to look at the log, or if there is no implementation to store the log, or what.
Isn't it enough to use the standard helm command then? I think.

I'm sick of Helm Extention, but I'm still trying hard to use it now, since I can raise an issue and get it fixed relatively aggressively.
However, I have to think of alternatives, so I'm looking into them in parallel.
So I found the helmfile project, and since this is a helm-focused project, I found it has a lot more documentation than helm Extention. Many of the documents are still outdated, but better than nothing.
Moreover, the helmfile project can also distribute kustomize and standard manifest files.

I was able to implement helmfile as an alternative to helm Extention by putting in the following manifest, although it is rough.
Please note that the permissions are not the best, so please refer to it for reference.
Also, the execution is a cronjob at the moment, but that's because it's too much work to create the logic to detect and execute ConfigMap updates.
If I ever give up on the helm Extension, I'd like to implement automatic execution in conjunction with the Reloader project. There may be a better way, but I can't think of it.

```
apiVersion: v1
kind: Namespace
metadata:
  name: helmfile
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: cluster-admin-bind
subjects:
- kind: ServiceAccount
  name: default
  namespace: helmfile
roleRef:
  kind: ClusterRole
  name: cluster-admin
  apiGroup: rbac.authorization.k8s.io
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: helmfile-config
  namespace: helmfile
data:
  helmfile.yaml: |
    helmDefaults:
      wait: true
      waitForJobs: true
      timeout: 1800
    repositories:
    - name: stable
      url: https://charts.helm.sh/stable
    - name: prometheus-community
      url: https://prometheus-community.github.io/helm-charts
    releases:
    - name: kube-prometheus-stack
      chart: prometheus-community/kube-prometheus-stack
      disableValidation: true
      namespace: monitoring
---
apiVersion: batch/v1
kind: CronJob
metadata:
  name: helmfile-executor
  namespace: helmfile
spec:
  schedule: "*/5  *  *  *  *"
  concurrencyPolicy: Forbid
  jobTemplate:
    spec:
      backoffLimit: 6
      template:
        spec:
          containers:
          - name: helmfile
            image: quay.io/roboll/helmfile:v0.143.0
            command:
            - /bin/bash
            - -c
            args:
            - "helmfile apply"
            volumeMounts:
            - mountPath: /helmfile.d/
              name: helmfile-config
          volumes:
          - name: helmfile-config
            configMap:
              name: helmfile-config
          restartPolicy: OnFailure
```

I hope k0s will reconsider whether it is best for them to develop their own integration with helm and kustomize, or to integrate with an external project like helmfile.
Personally, I think that k0s helm extention implemented as k8s CRD is more k8s native and better conceptualized than helmfile, but as I said before, I think it lacks development resources.
If you think it's just an extra and you don't want to spend too much development resources on it, I think everyone would be happier if you collaborated with an external project.
